### PR TITLE
EW-524 providing get personenkontexte endpoint

### PR DESCRIPTION
**Pull request by psachmann**
# Description
This PR contains the implementation for the Get/ personenkontexte endpoint and some refactoring's:
- Adding documentation for how we handle mapping
- Adding concrete types to the AutoMap for many decorators if the property is not Boolean, Number or String
- Better separation between dtos and response types in use cases and controllers
- Adding a settings.json.default for preconfigured GitHub Copilot settings
- Correcting routes for organization and person

## Links to Tickets or other pull requests
- <https://ticketsystem.dbildungscloud.de/browse/EW-524>

## Changes
See description and Jira ticket.

## Data security
Not affected.

## Deployment
Not affected.

## New Repos, NPM packages or vendor scripts
Not affected.

## Approval for review
- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

File: src/modules/person/domain/personenkontext.service.ts [hunk 1]
```diff
@@ -35,4 +35,13 @@ export class PersonenkontextService {
 
         return { ok: true, value: personenkontexte };
     }
+
+    public async findById(id: string): Promise<Result<PersonenkontextDo<true>, DomainError>> {
```
Comment (Fshmit) [inline src/modules/person/domain/personenkontext.service.ts:39]: maybe rename the method to findPersonKontextById to match the method names of the other services? see organisation and person services.

File: src/modules/person/api/personenkontext.uc.ts [hunk 2]
```diff
@@ -47,10 +53,32 @@ export class PersonenkontextUc {
...
-        const personenkontexte: PersonenkontextResponse[] = result.value.map(
-            (personenkontext: PersonenkontextDo<true>) =>
-                this.mapper.map(personenkontext, PersonenkontextDo, PersonenkontextResponse),
+        const personenkontexte: PersonenkontextDto[] = result.value.map((personenkontext: PersonenkontextDo<true>) =>
+            this.mapper.map(personenkontext, PersonenkontextDo, PersonenkontextDto),
         );
         return personenkontexte;
     }
+
+    public async findById(dto: FindPersonenkontextByIdDto): Promise<PersonendatensatzDto> {
```
Comment (Fshmit) [inline src/modules/person/api/personenkontext.uc.ts:62]: name of the method, same issue as in the service.

File: src/modules/person/api/personenkontext.controller.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    @ApiNotFoundResponse({ description: 'The personenkontext was not found.' })
+    @ApiForbiddenResponse({ description: 'Insufficient permissions to perform operation.' })
+    @ApiInternalServerErrorResponse({ description: 'An internal server error occurred.' })
+    public async findById(@Param() params: FindPersonenkontextByIdParams): Promise<PersonendatensatzResponse> {
+        const request: FindPersonenkontextByIdDto = this.mapper.map(
+            params,
+            FindPersonenkontextByIdParams,
+            FindPersonenkontextByIdDto,
+        );
+        const result: PersonendatensatzDto = await this.personenkontextUc.findById(request);
```
Comment (Fshmit) [inline src/modules/person/api/personenkontext.controller.ts:45]:
```suggestion
           try {
            const result: PersonendatensatzDto = await this.personenkontextUc.findById(request);
            const response: PersonendatensatzResponse = this.mapper.map(
                result,
                PersonendatensatzDto,
                PersonendatensatzResponse,
            );
            return response;
        } catch (error) {
            if (error instanceof EntityNotFoundError) {
                throw new HttpException(error.message, HttpStatus.NOT_FOUND);
            }
            throw error;
        };
```

**Review by Fshmit — CHANGES_REQUESTED** [review]
see comments

File: src/modules/person/api/personenkontext.controller.ts (hunk 3 above)
Comment (psachmann) [reply src/modules/person/api/personenkontext.controller.ts:45]: Done.

File: src/modules/person/api/personenkontext.uc.ts (hunk 2 above)
Comment (psachmann) [reply src/modules/person/api/personenkontext.uc.ts:62]: Done.

File: src/modules/person/domain/personenkontext.service.ts (hunk 1 above)
Comment (psachmann) [reply src/modules/person/domain/personenkontext.service.ts:39]: Done.

**Review by Fshmit — APPROVED** [review]

**Pull request merged**